pypsa/definitions/components.py

```diff
@@ -1,120 +1,126 @@
-            raise DeprecationWarning(
-                "The 'ind' attribute is deprecated and will be removed in future versions."
-            )
+    description: str
+    category: str
+    defaults: pd.DataFrame
+    standard_types: pd.DataFrame | None = None
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, ComponentTypeInfo):
+            return NotImplemented
 
-    def __repr__(self) -> str:
         return (
-            f"Component(name={self.name!r}, list_name={self.list_name!r}, "
-            f"attrs=Keys({list(self.attrs.keys())}), static=DataFrame(shape={self.static.shape}), "
-            f"dynamic=Keys({list(self.dynamic.keys())}))"
+            self.name == other.name
+            and self.list_name == other.list_name
+            and self.description == other.description
+            and str(self.category) == str(other.category)
+            and self.defaults.equals(other.defaults)
         )
 
-    # @deprecated(
-    #     deprecated_in="0.32",
-    #     removed_in="1.0",
-    #     details="Use `c.static` instead.",
-    # )
+    def __repr__(self) -> str:
+        return self.name + " Component Type"
+
     @property
-    def df(self) -> pd.DataFrame:
-        return self.static
-
-    # @deprecated(
-    #     deprecated_in="0.32",
-    #     removed_in="1.0",
-    #     details="Use `c.dynamic` instead.",
-    # )
+    @deprecated(
+        deprecated_in="0.32.0",
+        details="Use the 'category' attribute instead.",
+    )
+    def type(self) -> str:
+        return self.category
+
     @property
-    def pnl(self) -> Dict:
-        return self.dynamic
+    @deprecated(
+        deprecated_in="0.32.0",
+        details="Use the 'defaults' attribute instead.",
+    )
+    def attrs(self) -> pd.DataFrame:
+        return self.defaults
+
+
+class ComponentsStore(dict):
+    def __setattr__(self, name: str, value: Any) -> None:
+        if hasattr(ComponentsStore, name):
+            msg = "'ComponentsStore' object attribute " f"'{name}' can not be set."
+            raise AttributeError(msg)
+        self[name] = value
+
+    def __getattr__(self, item: str) -> Any:
+        """
+        Get attribute from the dictionary.
+
+        Examples
+        --------
+        >>> components = ComponentsStore()
+        >>> components["generator"] = Generators()
+        >>> components.generators
+        """
+        try:
+            return self[item]
+        except KeyError:
+            msg = f"'{self.__class__.__name__}' object has no component '{item}'."
+            raise AttributeError(msg)
+
+    def __getitem__(self, item: str | Sequence) -> Any:
+        """
+        Index single and multiple items from the dictionary.
+
+        Similar behavior to pandas.DataFrame.__getitem__.
+
+        Examples
+        --------
+        >>> components = ComponentsStore()
+        >>> components["generator"] = Generators()
+        >>> components["storage"] = Storage()
```

Comment:
Shouldn't this be `StorageUnits()` and `Stores()`?

We also need some consistency for capitalization and plural/singular. So far, we had smallcaps+plural and capitalized+singular.